compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <vector>
+
+namespace
+{
+
+class Shape
+{
+  friend class SECursor;
+
+private:
+  struct ShapeElement
+  {
+    uint32_t value;
+    std::vector<char> tags;
+  };
+
+public:
+  void add_elem(int v)
+  {
+    ShapeElement se;
+    se.value = v;
+    elements.push_back(ShapeElement(se));
+  }
+
+  void add_elem_with_tag(int v, char tag)
+  {
+    ShapeElement se;
+    se.value = v;
+    se.tags.push_back(tag);
+    elements.push_back(ShapeElement(se));
+  }
+
+  void add_elem(ShapeElement e) { elements.push_back(e); }
+
+  std::vector<ShapeElement> elements;
+};
+
+class SECursor
+{
+private:
+  uint32_t _start_idx; // [start_idx, _end_idx) points Shape.elements range
+  uint32_t _end_idx;
+  uint32_t _product; // product of [se[_start_idx], se[_end_idx -1]]
+
+  std::vector<Shape::ShapeElement> &se;
+
+public:
+  SECursor(Shape &s) : _start_idx(0), _end_idx(1), se(s.elements)
+  {
+    _product = s.elements.at(0).value;
```

Comment:
this assumes there is at least one element in `s.elements`.
I'm not sure what happens when is not not met.
maybe add some assertion?